src/services/shardQuerySplitting.ts

```diff
@@ -208,79 +225,56 @@ function splitQueriesByStreamShard(
   return response;
 }
 
-function groupShardRequests(shards: number[], range: TimeRange) {
-  const hours = range.to.diff(range.from, 'hour');
+function updateGroupSizeFromResponse(response: DataQueryResponse, currentSize: number) {
+  if (!response.data.length) {
+    // Empty response, increase group size
+    return currentSize + 1;
+  }
 
-  // Spread low and high volume around
-  shards = spreadSort(shards);
-  console.log(`Querying ${shards.join(', ')} shards`);
+  const metaExecutionTime: QueryResultMetaStat | undefined = response.data[0].meta?.stats?.find(
+    (stat: QueryResultMetaStat) => stat.displayName === 'Summary: exec time'
+  );
 
-  const maxRequests = calculateMaxRequests(shards.length, hours);
-  const groupSize = Math.ceil(shards.length / maxRequests);
-  const requests: number[][] = [];
+  if (metaExecutionTime) {
+    debug(`${metaExecutionTime.value}`);
+    // Positive scenarios
+    if (metaExecutionTime.value < 1) {
```

Comment:
What's the logic behind these value choices here? How do you feel about them being static, and needing to ship a new version of the plugin to change them?

Reply:
Good question. It has been defined by observation of the time that querying each shard take. It's basically a better heuristic to grup based on a signal (execution time).
If we see the need, we can record the slowest queries and have these values be a factor of it.